sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -364,11 +364,24 @@ class Analyzer(
 
         val (resolvedOrdering, missing) = resolveAndFindMissing(ordering, a, groupingRelation)
 
-        if (missing.nonEmpty) {
+        val addForAlias = new ArrayBuffer[NamedExpression]()
+        val aliasedOrdering = resolvedOrdering.zipWithIndex.map {
+          case (o, i) => {
+            o transform {
+              case aggOrSub @ (_: AggregateExpression | _: Substring) =>
```

Comment:
We don't need to add the aliased ordering to projection list if it's already there, right? I think we should filter out expressions which are already in `aggs`.